torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -94,7 +94,12 @@ def _reduce_value(
         assert self.mask_buffer.data is not None
 
         # apply the mask to the tensor that pending reduction
-        tensor[self.mask_buffer.data, :] = 0.0
+        # NOTE: Depending on the use case (gather op or embedding op),
+        # tensor dim can be equal or larger than the mask dim, respectively.
+        if tensor.ndim == self.mask_buffer.data.ndim:
+            tensor[self.mask_buffer.data] = 0.0
+        else:
```

Comment:
I think the main reason here is that the embedding output would produce an additional dimension compare to the input, hence the output masking logic become different? maybe have more clarification in the comment